### PR TITLE
Makefile: scripts: Add build args for proxy when using docker build

### PR DESCRIPTION
**Pull request by GabyCT**
This PR adds build args for proxy when using docker build, this is specially needed when we are behind a proxy to avoid failures.

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thanks @GabyCT!

**Review by stevenhorsman — COMMENTED** [review]
The code LGTM, but one question about where the `http_proxy` and `https_proxy` values are set - is the expectation that these environment variables are set on the self-hosted runner independently of the operator workflow?

**Comment by GabyCT** [comment]
@stevenhorsman changes applied

File: Makefile [hunk 1]
```diff
@@ -143,9 +146,9 @@ run: manifests generate fmt vet ## Run a controller from your host.
 docker-build: test ## Build docker image with the manager.
 ifneq (, $(PEERPODS))
 	@echo PEERPODS is enabled
-	docker build -t ${IMG} -f Dockerfile.peerpods .
+	docker build --build-arg http_proxy=$(http_proxy) --build-arg https_proxy=$(https_proxy) -t ${IMG} -f Dockerfile.peerpods .
```
Comment (mythi) [inline Makefile:149]: is there a reason why this approach was chosen over the better alternative which just setting the `~/.docker/config.json`?

Comment (GabyCT) [reply Makefile:149]: @mythi I have the proxies at the config but seems that when building the dockerfile is still failing without those arguments

**Comment by GabyCT** [comment]
@stevenhorsman now the CI for tdx is passing

**Review by stevenhorsman — APPROVED** [review]
Thanks @GabyCT

File: Makefile (hunk 1 above)
Comment (mythi) [reply Makefile:149]: I don't prefer this approach. It'd be good to understand *why* the best option does not work

**Pull request merged**